x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/WelfordAlgorithm.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.aggregation;
+
+/**
+ * Algorithm for calculating standard deviation, one value at a time
+ *
+ * @see <a href="https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Welford's_online_algorithm">
+ *         Welford's_online_algorithm</a> and
+ *         <a href="https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Parallel_algorithm">
+ *         Parallel algorithm</a>
+ */
+public final class WelfordAlgorithm {
```

Comment:
Should we fold this class into the StdDeviationStates#SingleState?

Reply:
I wasn't sure if we wanted to use it elsewhere (like if we wanted to support variance or have both sample and population standard deviation)?

Reply:
We can leave it as is.

Reply:
Feels like the kind of thing that could move moved to a common place like `libs/`, but while there is only one usage, it might as well stay here for now.